STACpopulator/api_requests.py

```diff
@@ -107,8 +107,12 @@ def post_stac_item(
         if update:
             LOGGER.info(f"Item {item_id} already exists. Updating.")
             r = session.put(os.path.join(stac_host, f"collections/{collection_id}/items/{item_id}"), json=json_data)
-            r.raise_for_status()
+            return f"Requests: {r.reason}"
+            # r.raise_for_status()
         else:
-            LOGGER.info(f"Item {item_id} already exists.")
+            LOGGER.warn(f"Item {item_id} already exists.")
     else:
-        r.raise_for_status()
+        return f"Requests: {r.reason}"
+        # r.raise_for_status()
```

Comment:
I think it is more appropriate to raise the exception and let the parent code handle it accordingly.
For example, a 403 would make all operations useless, while 409 could be ok if override was not requested, and 502 could indicate that the endpoint is wrong entirely. 
This information is loss with this strategy.